http-api/src/lib.rs

```diff
@@ -316,8 +329,25 @@ async fn remote_handler(
     Ok(warp::reply::json(&response))
 }
 
-async fn commits_handler(ctx: Context, project: Urn, sha: One) -> Result<impl Reply, Rejection> {
-    let reference = Reference::head(Namespace::from(project), None, sha);
+async fn history_handler(
+    ctx: Context,
+    project: Urn,
+    qs: HashMap<String, String>,
+) -> Result<impl Reply, Rejection> {
+    let mut fallback_to_head = false;
+    let sha = match qs.get("from").cloned() {
```

Comment:
You can set both vars here, it's nicer:
`let (sha, fallback_to_head) = match ...`